coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala

```diff
@@ -1196,67 +1196,62 @@ class PlannerHierarchySpec extends AnyFunSpec with Matchers with PlanValidationS
     validatePlan(execPlan1, expectedPlan1)
   }
 
-  it ("should pushdown aggregations/binary-joins when leaf plans are split across partitions") {
```

Comment:
Personally I find validating the query plan better than programatic checks, its easier to visualize how a plan should look like when expressed as string plans. That's precisely the objective of ``PlannerHierarchySpec`` when it was originally developed.

Reply:
But with this approach, there is zero (okay-- "substantially less") mental overhead to confirm all the tests look good. As long as you agree that:
```
    case class Test(query: String, lookbackSec: Long = WindowConstants.staleDataLookbackMillis / 1000, offsetSec: Long = 0) {
       def getExpectedRangesSec(): Seq[(Long, Long)] = {
         val snappedSecondStart = LogicalPlanUtils.snapToStep(timestamp = splitSec + offsetSec + lookbackSec,
                                                              step = stepSec,
                                                              origin = startSec)
         Seq((startSec, splitSec + offsetSec),
             (snappedSecondStart, endSec))
       }
     }
```
gives the correct ranges, then there's no mental math for each test (`split + offset + lookback` snapped to a periodic step). Plus, these trees all have the same simple structure, so programmatically validating everything doesn't need any complicated logic. Double-plus: `ExecPlan`s don't need to be re-printed whenever someone adds/removes/changes a plan field that was probably irrelevant to the test.

(I'll add a commit that lays these out as string-validations, though, since I expect my opinion is unpopular, and in the long-run it makes no real difference.)

Reply:
Touché-- I found a bug with the string-comparison method 🤦